### PR TITLE
JDB-001 create client table

### PR DESCRIPTION
**Pull request by dana-kainos**

### CODE REVIEW

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE `client` (
+    id int AUTO_INCREMENT NOT NULL,
+    name nvarchar(50) NOT NULL,
+    address nvarchar(50) NOT NULL,
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: Is this a normalised field? What about street address, postcode, county, country. Which do we need to store and how do we need to store it.

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE `client` (
+    id int AUTO_INCREMENT NOT NULL,
+    name nvarchar(50) NOT NULL,
+
+    # Address
+    street1 nvarchar(50) NOT NULL,
+    street2 nvarchar(50),
+    town nvarchar(50),
+    city nvarchar(50),
+    zip nvarchar(10) NOT NULL,
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: 'I live in ireland, i dont have a zip code'
'Kainoos is a uk organisation, whats zip'?

Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: Every address has a town and assigned city, we should accept, one, either or both

**Review by lukemKainos — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V1__create_client_table.sql (hunk 2 above)
Comment (dana-kainos) [reply src/main/resources/db/migration/V1__create_client_table.sql]: 'I live in ireland, in the middle of nowhere in the gaeltacht, i don't have an assigned town, only a postal district at city level'

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE `client` (
+    id int AUTO_INCREMENT NOT NULL,
+    name nvarchar(50) NOT NULL,
+
+    # Address
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: why is there a comment here, db field names should be self explanatory e.g. addressLine1, addressLine2 is a common standard along with the rest of the fields. We don't need this comment

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    street1 nvarchar(50) NOT NULL,
+    street2 nvarchar(50),
+    town nvarchar(50),
+    city nvarchar(50),
+    postcode nvarchar(10),
+    country char(2) NOT NULL
+
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: why is there a space here?

**Review by lukemKainos — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V1__create_client_table.sql [hunk 5]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE `client` (
+    id int AUTO_INCREMENT NOT NULL,
```
Comment (lukemKainos) [inline src/main/resources/db/migration/V1__create_client_table.sql]: This cant be called id, it should be called client id. It makes it hard to read should we have a foreign key

**Review by lukemKainos — APPROVED** [review]

**Review by dylanmckk — APPROVED** [review]

**Pull request merged**